CTP/common/ext/run_shell.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+# 
+# Copyright (c) 2016, Search Solution Corporation. All rights reserved.
+# 
+# Redistribution and use in source and binary forms, with or without 
+# modification, are permitted provided that the following conditions are met:
+# 
+#   * Redistributions of source code must retain the above copyright notice, 
+#     this list of conditions and the following disclaimer.
+# 
+#   * Redistributions in binary form must reproduce the above copyright 
+#     notice, this list of conditions and the following disclaimer in 
+#     the documentation and/or other materials provided with the distribution.
+# 
+#   * Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products 
+#     derived from this software without specific prior written permission.
+# 
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+# INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE 
+# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+#
+
+
+role=""
+tmplog=""
+feedback_type="database"
+coverage_controller_target_dir=""
+coverage_collaborate_url=""
+is_continue_mode=$1
+shell_config_template=""
+shell_fm_test_conf="${CTP_HOME}/conf/shell_runtime.conf"
+
+coverage_controller_ip=$MKEY_COVERAGE_UPLOAD_IP
+coverage_controller_user=$MKEY_COVERAGE_UPLOAD_USER
+coverage_controller_pwd=$MKEY_COVERAGE_UPLOAD_PWD
+coverage_controller_port=$MKEY_COVERAGE_UPLOAD_PORT
+testcase_timeout=`getMsgValue $MKEY_TESTCASE_TIMEOUT 7200`
```

Comment:
Please remove 
+testcase_timeout=`getMsgValue $MKEY_TESTCASE_TIMEOUT 7200`
+fail_case_retry_count=`getMsgValue $MKEY_MAX_RETRY_COUNT 0`
+db_charset=`getMsgValue $MKEY_TESTING_DEFAULT_CHARSET en_US`

It's better to configure them in shell config file.